### ISSUE TITLE
Stabilize and document API response schemas

### ISSUE THREAD

**Issue by pmatseykanets** [sdwolfe32/trumail#43]
It would be great if it'd be possible to stabilize, document and freeze schemas for successful and error API responses at least per version.

There were a few changes lately that broke API consumers. I.e. 

- hostExists field was removed and then reinstated a few days later
- gravatar field removed
- the error response changed

Thank you so much for your effort, time and creating a great service.

**Comment by sdwolfe32** [comment]
Yeah I've felt a little funny about doing that lately. I didn't mean to break anyone consuming the api but the schema now should be the final format for v1. The past week I've been trying to standardize and simplify just SMTP verification for in order to have a solid v1 format. I apologize for making your life difficult! I'll make sure not to make another formatting change until we move to v2

**Comment by sdwolfe32** [comment]
As a matter of fact I just noticed the error format is currently not how I wanted it. It failed over to the format from the http framework I'm using bc of a recent refactor. I'm changing it back in a few. Again, sorry about that

**Comment by sdwolfe32** [comment]
Again I really apologize for the error mismatch bug. The fix should be in production now. Again I won't change any formatting for the API v1 output and will move to v2 if there are changes.
